krm-functions/interface-fn/design.md

```diff
@@ -0,0 +1,157 @@
+Interface management KRM function design
+========================================
+
+Purpose of the function
+-----------------------
+
+The `Interface` KRM function is designed to be a part of the pipeline of a Nephio NF kpt package.
+
+Its primary purpose is to find all `Interface` custom resources in the package and expand it to a set of more specific (child) KRM resources, i.e. `IPAllocation` and `VLANAllocation` resources. 
+
+The function is also designed to be part of the "`Condition` choreography" that is meant to synchronize the effects of multiple KRM functions acting on the same kpt package. It adds `Conditions` to the `Status` of the `Kptfile` object indicating that the `IPAllocation` and `VLANAllocation` requests are not fulfilled yet. The function also triggers the addition of a `NetworkAttachmentDefinition` later by a "NAD generator" KRM function by setting the corresponding `Condition` to false.
+
+
+Child resources generated for an `Interface` resource
+-------------------------------------------------------
+
+For each `Interface` the function will generate one `IPAllocation` and one `VLANAllocation` CR. In order to generate the `Spec` of those child resources it uses information also from the `ClusterContext` resource that the kpt package assumed to also contain.
+
+Let's see an example! Assuming that a kpt package contains the following two resources (among others):
+
+    apiVersion: req.nephio.org/v1alpha1
+    kind: Interface
+    metadata:
+        name: n6
+        annotations:
+            config.kubernetes.io/local-config: "true"
+    spec:
+        networkInstance:
+            name: vpc-internet
+        cniType: sriov
+        attachementType: vlan
+    ---
+    apiVersion: infra.nephio.org/v1alpha1
+    kind: ClusterContext
+    metadata: 
+        name: cluster-context
+        annotations:
+            config.kubernetes.io/local-config: "true"
+    spec:
+        cniConfig:
+            cniType: macvlan
+            masterInterface: eth1
+        siteCode: edge1
+        region: us-central1
+
+the following list of CRs will be added to package:
+
+    apiVersion: ipam.nephio.org/v1alpha1
+    kind: IPAllocation
+    metadata:
+      name: n6-ip-abcd
```

Comment:
Can we also include the following as part of metadata:
As compliant with:- https://docs.google.com/document/d/1wAgPO0IDyN-UqpB3twv3e-9mumck-5o3j8f74FLkKPI/ 
```
  labels:
    nephio.org/interface: n3
```

If not, is there a field that has only interface details?